manticore/native/manticore.py

```diff
@@ -27,6 +27,7 @@ def __init__(self, path_or_state, argv=None, workspace_url=None, policy='random'
             if not os.path.isfile(path_or_state):
                 raise OSError(f'{path_or_state} is not an existing regular file')
 
+            self._binpath = path_or_state
```

Comment:
Hmm... for `Linux` we already have the binary path in `self.state.platform.program`, just for clarification:
* `_make_initial_path` creates a given `Platform` object - `Linux`
* for `Linux`: https://github.com/trailofbits/manticore/blob/master/manticore/platforms/linux.py#L404
* the `State` itself has a `_platform` attribute and `platform` property - https://github.com/trailofbits/manticore/blob/master/manticore/core/state.py#L82

But this is not true for `Decree` - we should probably add this property there. Or maybe the `class Platform` should assign `self._program_path` or sth.

Reply:
I can first set a `path` property in `class Platform`:

```python
class Platform(Eventful):
    def __init__(self, path, **kwargs):
        self.path = path
        super().__init__(**kwargs)
```

and then assign `self._binpath`to `initial_state.platform.path`. Since DECREE binaries can't be resolved through this feature, calling `resolve()` will throw a `NotImplementedError`.

Reply:
I'm not sure about having this assigned twice. Maybe we should have a private property instead e.g. in `ManticoreBase` that would resolve to `self._initial_state.platform.path`?

On the other hand - if a Linux binary calls `execve` with another path, that property will still resolve to the old/first program path.

@yan @mossberg any comments on this?

Reply:
> Maybe we should have a private property instead e.g. in ManticoreBase that would resolve to self._initial_state.platform.path?

i like this solution :) the execve issue is esoteric enough i think that we can maybe just put a comment around our sys_execve model, and leave it.